### PR TITLE
Added more template elements and callback for file size

### PR DESCRIPTION
**Pull request by scsides**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by krlberry — APPROVED** [review]
Looks good! Thoughts about putting comments in the xml label template to explain the ` - 1` and `- 2`? Good/useful documentation? Overkill?

**Comment by scsides** [comment]
one -1 is because it is 1 based not zero, the other -1 is because it is the start byte of the cube data and not the number of bytes in the header

File: isis/src/base/apps/topds4/topds4.cpp [hunk 1]
```diff
@@ -30,6 +31,9 @@ namespace Isis {
     Process p;
     p.SetInputCube(icube);
 
+    // NEED TO WRITE AND CLOSE THE OUTPUT FILE BEFORE RENDERING SO THE FILE SIZE CALLBACK CAN GET THE FINAL FILE SIZE
+    QString outputFile = ui.GetFileName("TO");
```
Comment (jessemapel) [inline isis/src/base/apps/topds4/topds4.cpp:35]: I thought this was the size of the image file, not the pds4 label?

**Review by jessemapel — APPROVED** [review]

File: isis/src/base/apps/topds4/topds4.cpp (hunk 1 above)
Comment (jessemapel) [reply isis/src/base/apps/topds4/topds4.cpp:35]: This is the output copy of the cube per other discussion

**Pull request merged**